ecl/hthor/hthor.cpp

```diff
@@ -457,22 +469,38 @@ void CHThorDiskWriteActivity::stop()
         uncompressedBytesWritten = outSeq->getPosition();
     close();
     updateWorkUnitResult(numRecords);
-    if((helper.getFlags() & (TDXtemporary | TDXjobtemp) ) == 0 && !agent.queryResolveFilesLocally())
+    if((helperFlags & (TDXtemporary | TDXjobtemp) ) == 0 && !agent.queryResolveFilesLocally())
         publish();
     incomplete = false;
     if(clusterHandler)
         clusterHandler->finish(file);
     CHThorActivityBase::stop();
-    if (helper.getFlags() & TDXvarfilename)
+    if (helperFlags & TDXvarfilename)
         filename.clear();
 }
 
 void CHThorDiskWriteActivity::resolve()
 {
     OwnedRoxieString rawname = helper.getFileName();
-    mangleHelperFileName(mangledHelperFileName, rawname, agent.queryWuid(), helper.getFlags());
+    mangleHelperFileName(mangledHelperFileName, rawname, agent.queryWuid(), helperFlags);
     assertex(mangledHelperFileName.str());
-    if((helper.getFlags() & (TDXtemporary | TDXjobtemp)) == 0)
+
+    if (useGenericReadWrites)
+    {
+        // Merge options from helper into providerOptions and formatOptions if required
+        if (helperFlags & TDXdynprovideroptions)
+        {
+            CPropertyTreeWriter providerOptsWriter(providerOptions);
```

Comment:
This will need to be called once if TDXdynprovideroptions is not set, but may be context dependent if TDXdynprovideroptions is set.  Revisit in a later PR.